core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -314,100 +161,36 @@ public QueryBuilder parseInnerFilterToQueryBuilder(String queryName) throws IOEx
         }
     }
 
-    /**
-     * @deprecated replaced by calls to parseInnerFilterToQueryBuilder(String queryName) for the resulting queries
-     */
-    @Nullable
-    @Deprecated
-    public Query parseInnerFilter(String queryName) throws IOException, QueryParsingException {
-        QueryBuilder builder = parseInnerFilterToQueryBuilder(queryName);
-        return (builder != null) ? builder.toQuery(this) : null;
-    }
-
-    public Collection<String> simpleMatchToIndexNames(String pattern) {
-        return indexQueryParser.mapperService.simpleMatchToIndexNames(pattern, getTypes());
-    }
-
-    public MappedFieldType fieldMapper(String name) {
-        return failIfFieldMappingNotFound(name, indexQueryParser.mapperService.smartNameFieldType(name, getTypes()));
-    }
-
-    public ObjectMapper getObjectMapper(String name) {
-        return indexQueryParser.mapperService.getObjectMapper(name, getTypes());
+    public boolean isFilter() {
```

Comment:
I have the feeling this should not be handled in parse phase but only in toQuery....